server/service/individualServices/ReadInventoryData.js

```diff
@@ -1,28 +1,103 @@
+ *    to gather installed-firmware list of given mount-name
+ * @param {String} mountName Identifier of the device at the Controller
+ * @param {Object} requestHeaders Holds information of the requestHeaders like Xcorrelator , CustomerJourney,User etc.
+ * @param {Integer} traceIndicatorIncrementer traceIndicatorIncrementer to increment the trace indicator
+ * @returns {Object} return values of installed-firmware list and traceIndicatorIncrementer
+ */
+  async function RequestForProvidingAcceptanceDataCausesReadingFirmwareList(mountName, requestHeaders, traceIndicatorIncrementer) {
+    let installedFirmwareResponse = {};
+    let installedFirmwareList = [];
+    let pathParamList = [];
+    const fcNameForReadingFirmwareList = "RequestForProvidingAcceptanceDataCausesReadingFirmwareList";
+    const stringNameForReadingFirmwareList = "RequestForProvidingAcceptanceDataCausesReadingFirmwareList.FirmwareFromCache";
+    try {
+      let clientAndFieldParamsForFirmwareList = await IndividualServiceUtility.getConsequentOperationClientAndFieldParams(fcNameForReadingFirmwareList, stringNameForReadingFirmwareList);
+      pathParamList.push(mountName);
+      /****************************************************************************************************
+      * process required attributes from firmware-collection
+      *   RequestForProvidingAcceptanceDataCausesReadingFirmwareList
+      *     MWDI://core-model-1-4:network-control-domain=cache/control-construct={mount-name}/firmware-1-0:firmware-collection
+      *****************************************************************************************************/
+      let response = await IndividualServiceUtility.forwardRequest(clientAndFieldParamsForFirmwareList, pathParamList, requestHeaders, traceIndicatorIncrementer++);
+      if (Object.keys(response).length > 0) {
+        let firmwareComponentList = response[FIRMWARE.MODULE + FIRMWARE.COLLECTION][FIRMWARE.COMPONENT_LIST];
+        for (let i = 0; i < firmwareComponentList.length; i++) {
+          let installedFirmware = {};
+          let firmwareComponentPac = firmwareComponentList[i][FIRMWARE.PAC];
+          let firmwareComponentCapability = firmwareComponentPac[FIRMWARE.CAPABILITY];
+          let firmwareComponentClass = firmwareComponentCapability[FIRMWARE.CLASS];
+          let expectedFirmwareComponentClass = FIRMWARE.MODULE + FIRMWARE.CLASS_TYPE;
+          if (firmwareComponentClass == expectedFirmwareComponentClass) {
+            installedFirmware.firmwareComponentName = firmwareComponentCapability[FIRMWARE.NAME];
+            installedFirmware.firmwareComponentVersion = firmwareComponentCapability[FIRMWARE.VERSION];
+            installedFirmware.firmwareComponentStatus = firmwareComponentPac[FIRMWARE.STATUS][FIRMWARE.STATUS];
+            installedFirmwareList.push(installedFirmware);
+          }
+        }
+      }
+      installedFirmwareResponse = {
+        installedFirmware: installedFirmwareList,
```

Comment:
variable shall be  "installedFirmware**List** : installedFirmwareList" as it represents a list of installedFirmware